### ISSUE TITLE
Image Picker

### ISSUE THREAD

**Issue by soareseneves** [TheJuki/KFormMaster#119]
I have a custom Image Picker Element, I would like to make a pull request to add it to your source but there is a problem, I am using this library to select/crop the image: https://github.com/Dhaval2404/ImagePicker It's a great library but it only support API 19+ and yours supports API 15+

Anyway, the image element has a loading view that can be used for upload progress, here is how it looks like:

![imagepicker](https://user-images.githubusercontent.com/16960561/57453928-ff1a5c80-723d-11e9-9359-271cc2bb3660.jpg)

and here is a video:

https://drive.google.com/open?id=1O2OS_c1P33VWf11UsBDBsrKtFeEv2qNG

I can replace the "Dialog" I use with a simple AlertDialog and also add a few options like "Only Camera/Gallery Option"

What do you think?

**Comment by TheJuki** [comment]
Go ahead with your proposal. Thanks for all of the PRs! After this change I'll push the next release.

**Comment by soareseneves** [comment]
No problem, I will try to finish tomorrow and then create a PR!

**Comment by soareseneves** [comment]
Hey @TheJuki, the image element is almost ready, I'm only strugling with the dialog which is not styling as the other dialogs on the library (Dropdown, etc), as soon I get to fix it I will create the PR.

**Comment by soareseneves** [comment]
Alright @TheJuki, the element is ready, I tested everything I could remember, currently it's using my ImagePicker Library fork which has some fixes but the library author already commited my fixes and will bump his version in 3 hours, for now you can take a look and test here: https://github.com/soareseneves/KFormMaster
As soon Dhaval2404/ImagePicker is updated I will change the gradle file, commit and send a pull request.

I know my code is not the greatest so please change anything that is not on your code standards.

Here is an example of usage (lots of options haha)

```
imageView(1) {
    displayDivider = false
    //imageTransformation = CircleTransform(borderColor = Color.WHITE, borderRadius = 3) //Default value for this is CircleTransform(null) so it makes image round without borders
    required = false
    //theme = R.style.SWDarkDialog //This is to theme the default dialog when onClickListener is not used.
    value = "https://www.myimage.com/image.jpg" //(String) This needs to be an image URL or an image FILE (absolutePath)
    imagePickerOptions = {
        //This lets you customize the ImagePicker library, specifying Crop, Dimensions and MaxSize options
        it.cropX = 3f
        it.cropY = 4f
        it.maxWidth = 150
        it.maxHeight = 200
    }
    onSelectImage = { file ->
        //If file is null it means an error ocurred trying to select the image
        if (file != null) {
            //this.loadingView?.toggle(state = true) //this will reset the progressBar, show it and set to 0%

            //this.loadingView?.setProgressAndText(progress = 50f, decimalNumber = 0) // this sets the progress, the text and also the amount of decimal numbers to show, it's optional, you can just use the progress parameter, this is useful for an image upload!
            
            //this.loadingView?.toggle(state = false, fade = true) //this will "hide" the loading by setting the text to empty and the progress to 0, there is also a fade option to make it smothier!
            }
        } else {
            Toast.makeText(this@MyActivity, "Error getting the image", Toast.LENGTH_LONG).show()
        }
    }
}
```

Wait, there is more! if you don't want to use the custom AlertDialog to show the options (Camera, Gallery, Remove, Cancel), you can set the onClickListener property and create your own dialog/view or whatever!

For this example I am using https://github.com/WindSekirun/ActionSheet but you can use anything

```
imageView(1) {
    //...all other options
    onClickListener = {
        val config = ActionSheetConfig.Builder()
            .addItem("Camera", "Gallery")
            .addItem("Remove", Color.RED)
            .setCancelableOnTouchOutside(true)
            .setCancelButton("Cancel")
            .setOnActionButtonClickListener(OnActionButtonClickListener { actionSheet, actionButton, index ->
                if (index == 0 || index == 1){
                    openImagePicker(if (index == 0) ImageProvider.CAMERA else ImageProvider.GALLERY)
                }

                if (index == 2){
                    clearImage()
                }
            })
            .build()


        ActionSheet.show(this@MyActivity, this@MyActivity.supportFragmentManager, config)
    }
    //...all other options
}
```

Tell me what you think.

**Comment by soareseneves** [comment]
Image Picker Library has been updated! PR on the way.

**Comment by TheJuki** [comment]
I just released [6.4.0](https://github.com/TheJuki/KFormMaster/releases/latest). Please update your dependency and let me know if the new Image View element works for you. I cut out the loading code since it did not look great (the colored circle) and I would need to add more tests for that one class. Also, there is already a progress element that can be used for that. 

I need to add actual tests for the new Image View but I did update the examples and manually tested it.

**Comment by TheJuki** [comment]
I am closing this issue. If you have any issues feel free to reopen.
